### PR TITLE
Porting to erigon 2.32

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkrpc/commands/debug_api.cpp [hunk 1]
```diff
@@ -390,9 +390,10 @@ boost::asio::awaitable<void> DebugRpcApi::handle_debug_trace_block_by_number(con
         const auto block_with_hash = co_await core::read_block_by_number(*context_.block_cache(), tx_database, block_number);
 
         debug::DebugExecutor executor{*context_.io_context(), tx_database, workers_, config};
-        const auto debug_traces = co_await executor.execute(block_with_hash.block);
+        const auto debug_trace_result = co_await executor.execute_block(block_with_hash.block);
```
Comment (canepat) [inline silkrpc/commands/debug_api.cpp:393]: I suggest a different approach here:
- do not add the new method `DebugExecutor::execute_block`, but use the existing `DebugExecutor::execute`
- create a `DebugTraceResultList` object containing `debug_traces` returned by `execute`
- add `to_json` serialisation method for `DebugTraceResultList` and handle the additional `result` field there

File: silkrpc/commands/debug_api.cpp [hunk 2]
```diff
@@ -434,9 +435,9 @@ boost::asio::awaitable<void> DebugRpcApi::handle_debug_trace_block_by_hash(const
         const auto block_with_hash = co_await core::read_block_by_hash(*context_.block_cache(), tx_database, block_hash);
 
         debug::DebugExecutor executor{*context_.io_context(), tx_database, workers_, config};
-        const auto debug_traces = co_await executor.execute(block_with_hash.block);
+        const auto debug_trace_result = co_await executor.execute_block(block_with_hash.block);
```
Comment (canepat) [inline silkrpc/commands/debug_api.cpp:438]: Same as above

File: silkrpc/core/evm_debug.cpp [hunk 3]
```diff
@@ -80,6 +80,11 @@ void to_json(nlohmann::json& json, const DebugTrace& debug_trace) {
     }
 }
 
+void to_json(nlohmann::json& json, const DebugTraceResult& debug_trace_result) {
+    json["result"] = debug_trace_result.debug_trace;
+}
+
+
```
Comment (canepat) [inline silkrpc/core/evm_debug.cpp:87]: This will be removed using the new approach.

File: silkrpc/core/evm_debug.cpp [hunk 4]
```diff
@@ -293,6 +298,46 @@ boost::asio::awaitable<std::vector<DebugTrace>> DebugExecutor<WorldState, VM>::e
     co_return debug_traces;
 }
 
+template<typename WorldState, typename VM>
+boost::asio::awaitable<std::vector<DebugTraceResult>> DebugExecutor<WorldState, VM>::execute_block(const silkworm::Block& block) {
```
Comment (canepat) [inline silkrpc/core/evm_debug.cpp:302]: This will be removed using the new approach.

File: silkrpc/core/evm_debug.hpp [hunk 5]
```diff
@@ -121,7 +121,16 @@ struct DebugTrace {
     DebugConfig debug_config;
 };
 
+struct DebugTraceResult {
+    DebugTrace debug_trace;
+};
```
Comment (canepat) [inline silkrpc/core/evm_debug.hpp:126]: This will be removed using the new approach.

File: silkrpc/core/evm_debug.hpp [hunk 6]
```diff
@@ -121,7 +121,16 @@ struct DebugTrace {
...
+
+struct DebugTraceResultList {
+    std::vector<DebugTrace> debug_traces;
+};
+
 void to_json(nlohmann::json& json, const DebugTrace& debug_trace);
+void to_json(nlohmann::json& json, const DebugTraceResult& debug_trace_result);
```
Comment (canepat) [inline silkrpc/core/evm_debug.hpp:133]: This will be removed using the new approach and `to_json` for `DebugTraceResultList ` will be added

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkrpc/core/evm_debug.cpp (hunk 3 above)
Comment (lupin012) [reply silkrpc/core/evm_debug.cpp:92]: Code changes according the review comment

File: silkrpc/commands/debug_api.cpp (hunk 1 above)
Comment (lupin012) [reply silkrpc/commands/debug_api.cpp]: Ok

File: silkrpc/core/evm_debug.cpp (hunk 3 above)
Comment (lupin012) [reply silkrpc/core/evm_debug.cpp:92]: ok

File: silkrpc/core/evm_debug.cpp (hunk 4 above)
Comment (lupin012) [reply silkrpc/core/evm_debug.cpp]: ok

File: silkrpc/core/evm_debug.hpp (hunk 5 above)
Comment (lupin012) [reply silkrpc/core/evm_debug.hpp]: ok

File: silkrpc/core/evm_debug.hpp (hunk 6 above)
Comment (lupin012) [reply silkrpc/core/evm_debug.hpp]: ok

**Review by canepat — APPROVED** [review]
LGTM

**Pull request merged**